server/sonar-server/src/main/java/org/sonar/server/branch/pr/ws/ListAction.java

```diff
@@ -0,0 +1,140 @@
+      .setDescription("List the pull requests of a project.<br/>" +
+        "Requires 'Administer' rights on the specified project.")
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setHandler(this);
+
+    addProjectParam(action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String projectKey = request.mandatoryParam(PARAM_PROJECT);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto project = componentFinder.getByKey(dbSession, projectKey);
+      userSession.checkComponentPermission(UserRole.USER, project);
+      checkArgument(project.isEnabled() && PROJECT.equals(project.qualifier()), "Invalid project key");
+
+      List<BranchDto> pullRequests = dbClient.branchDao().selectByComponent(dbSession, project).stream()
+        .filter(b -> b.getBranchType() == PULL_REQUEST)
+        .collect(toList());
+      List<String> pullRequestUuids = pullRequests.stream().map(BranchDto::getUuid).collect(toList());
+
+      Map<String, BranchDto> mergeBranchesByUuid = dbClient.branchDao()
+        .selectByUuids(dbSession, pullRequests.stream().map(BranchDto::getMergeBranchUuid).filter(Objects::nonNull).collect(toList()))
+        .stream().collect(uniqueIndex(BranchDto::getUuid));
+      Map<String, BranchStatistics> branchStatisticsByBranchUuid = issueIndex.searchBranchStatistics(project.uuid(), pullRequestUuids).stream()
+        .collect(uniqueIndex(BranchStatistics::getBranchUuid, Function.identity()));
+      Map<String, String> analysisDateByBranchUuid = dbClient.snapshotDao().selectLastAnalysesByRootComponentUuids(dbSession, pullRequestUuids).stream()
+        .collect(uniqueIndex(SnapshotDto::getComponentUuid, s -> formatDateTime(s.getCreatedAt())));
+
+      ProjectPullRequests.ListWsResponse.Builder protobufResponse = ProjectPullRequests.ListWsResponse.newBuilder();
+      pullRequests
+        .forEach(b -> addPullRequest(protobufResponse, b, mergeBranchesByUuid, branchStatisticsByBranchUuid.get(b.getUuid()),
+          analysisDateByBranchUuid.get(b.getUuid())));
+      writeProtobuf(protobufResponse.build(), request, response);
+    }
+  }
+
+  private static void addPullRequest(ProjectPullRequests.ListWsResponse.Builder response, BranchDto branch, Map<String, BranchDto> mergeBranchesByUuid,
+    BranchStatistics branchStatistics, @Nullable String analysisDate) {
+    Optional<BranchDto> mergeBranch = Optional.ofNullable(mergeBranchesByUuid.get(branch.getMergeBranchUuid()));
+    ProjectPullRequests.PullRequest.Builder builder = ProjectPullRequests.PullRequest.newBuilder();
+    String branchKey = branch.getKey();
+    builder.setId(branchKey);
+    //TODO get the branch from PULL_REQUEST_DATA
+    builder.setBranch(branchKey);
+    //TODO get the URL from PULL_REQUEST_DATA
+    builder.setUrl("https://github.com/SonarSource/sonar-core-plugins/pull/32");
+    //TODO get the branch title from PULL_REQUEST_DATA
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fbranch%2Ffirst-class-pull-request&issues=AWGZgq6zqu6zmfejgj1z&open=AWGZgq6zqu6zmfejgj1z)

[ISSUE_KEY]: AWGZgq6zqu6zmfejgj1z